### PR TITLE
Make `set-aggregate-public-key` private

### LINKED ISSUE #4101: Nakamoto: make set-aggregate-public-key private in pox-4
Inside pox-4, there is a public function "set-aggregate-public-key". You don't need to remove this per se; but it should not be publicly reachable. As signers vote for their aggregate public key for a given reward cycle, after the 70% threshold has been reached, then this function can be called internally. However, it can also be made private prior to this signer change and the node call it at the start of every pox-4 reward cycle instead. This MUST happen in Neon because this function should not be publicly reachable.

### PR DESCRIPTION
**Pull request by xoloki**
### Description
Currently, the `set-aggregate-public-key` function in `pox-4` is public, but it needs to be private.  This will require that the key is set when a new reward cycle begins.

In the future this will only happen after `DKG` completes, but for now just set the key to that of the previous cycle.

### Applicable issues

- fixes #4101 

### Checklist

- [ ] Test coverage for new or modified code paths
- [ ] Changelog is updated
- [ ] Required documentation changes (e.g., `docs/rpc/openapi.yaml` and `rpc-endpoints.md` for v2 endpoints, `event-dispatcher.md` for new events)
- [ ] New clarity functions have corresponding PR in `clarity-benchmarking` repo
- [ ] New integration test(s) added to `bitcoin-tests.yml`

### CODE REVIEW

File: stackslib/src/chainstate/nakamoto/mod.rs [hunk 1]
```diff
@@ -2514,6 +2519,78 @@ impl NakamotoChainState {
...
+                        StacksAddress::burn_address(false).into(),
+                        None,
+                        LimitedCostTracker::Free,
+                        |vm_env| {
+                            vm_env.execute_contract_allow_private(
+                                &boot_code_id(POX_4_NAME, false),
+                                "set-aggregate-public-key",
+                                &vec![
+                                    SymbolicExpression::atom_value(Value::UInt(
+                                        my_reward_cycle as u128,
```
Comment (jferrant) [inline stackslib/src/chainstate/nakamoto/mod.rs:2582]: Might want to change this to u128::from(my_reward_cycle)

File: stackslib/src/chainstate/stacks/boot/pox-4.clar [hunk 2]
```diff
@@ -1332,7 +1332,7 @@
 ;; Set the aggregate public key to the provided value
 ;; TODO: https://github.com/stacks-network/stacks-core/issues/4101
```
Comment (jferrant) [inline stackslib/src/chainstate/stacks/boot/pox-4.clar:1333]: Don't forget to remove the TODO if its done!

File: stackslib/src/chainstate/nakamoto/mod.rs [hunk 3]
```diff
@@ -2514,6 +2519,78 @@ impl NakamotoChainState {
             tenure_height,
         )?;
 
+        if !block.is_first_mined() {
```
Comment (jcnelson) [inline stackslib/src/chainstate/nakamoto/mod.rs:2522]: What you have here is fine, but can you put it into its own method?  It'll make testing it easier.  Thanks! :pray:

Comment (kantai) [reply stackslib/src/chainstate/nakamoto/mod.rs:2522]: Yep -- I think this code block should be moved into its own function.

File: stackslib/src/chainstate/stacks/boot/pox-4.clar (hunk 2 above)
Comment (xoloki) [reply stackslib/src/chainstate/stacks/boot/pox-4.clar]: Fixed

File: stackslib/src/chainstate/nakamoto/mod.rs (hunk 1 above)
Comment (xoloki) [reply stackslib/src/chainstate/nakamoto/mod.rs]: Fixed

File: testnet/stacks-node/src/mockamoto/tests.rs [hunk 4]
```diff
@@ -129,7 +129,7 @@ fn observe_100_blocks() {
 }
 
 #[test]
```
Comment (wileyj) [inline testnet/stacks-node/src/mockamoto/tests.rs:131]:
```suggestion
#[test]
#[ignore]
```

**Comment by jferrant** [comment]
Replaced by https://github.com/stacks-network/stacks-core/pull/4165#pullrequestreview-1780071969

**Pull request closed without merging**